conan/tools/cmake/toolchain/blocks.py

```diff
@@ -1280,6 +1280,24 @@ def context(self):
                     self._toolchain.preprocessor_definitions.configuration_types}
 
 
+class OverriddenFindPackageBlock(Block):
+    @property
+    def template(self):
+        return textwrap.dedent("""\
+            # Override find_package to warn users when find_package will resolve to non Conan provided package
+            include(_conan_direct_list OPTIONAL)
+            macro(find_package _name)
+                _find_package(${_name} ${ARGN})
```

Comment:
This is a bit concerning to me. It used to be working but undocumented behavior of CMake.
Now, with the CMake dependency providers and the ``cmake-conan`` integration, it might have some unexpected interactions.

In understand the aim to have better diagnostics for non-conan packages, but this seems risky and a bit too much and out of the scope of the generated ``conan_toolchain.cmake``. At some point Conan cannot be the guardian of every possible error that can happen on the user side.

Reply:
Yup, undocumented (although widely relied on, so I'd say unlikely to break). I'm not concerned about `cmake-conan` because cmake-conan _only_ relies on the CMakeDeps generator, there's no opportunity for the generated `conan_toolchain.cmake` to be included at all (even if the `CMakeToolchain` generator is enabled).

I would consider this feature if this code snippet is _not_ generated by default, and only opt-in e.g. `tools.cmake.cmaketoolchain:trace_find_package_calls_experimental`

This feature is not "generally" useful for consumers or users, I'd say it's mostly useful for recipe authors (when adding a new version or authoring a new recipe) - it can be a (considerable) time saver to highlight (to the recipe author) that something may be wrong. May even be useful as an optionally enabled thing in Conan Center CI.

Reply:
I've added your suggestion to guard its creation based on an experimental conf